Carlos/DiskCacheLevel.swift

```diff
@@ -161,7 +159,7 @@ public class DiskCacheLevel<K: StringConvertible, T: NSCoding>: CacheLevel {
     }
   }
   
-  private func setDataSync(data: T, key: K) {
+  private func setDataSync(data: T, key: K, promise: Promise<()>) {
```

Comment:
Not sure if inlining would make much sense as this is a nice self-contained block of code, albeit with an injected promise after my change. I'd leave it as it is for now and then it can remind you about `updateAccessDate` later.